todos.js

```diff
@@ -180,6 +179,48 @@ var handler = {
 	}
 };
 
+let enterHandler = function(keyCode, functionToCall, parameters){
+	if(keyCode === 'Enter'){
+		//Need to bind handler for the 'this' error
+		functionToCall = functionToCall.bind(handler);
+		functionToCall(...parameters);
+	}
+};
+
+//Enter event handler for add input
+let todoAddTextInput = document.getElementById('todoAddTextInput');
+
+todoAddTextInput.addEventListener("keypress", function(e){
+	enterHandler(e.key, handler.addTodos, []);
+});
+
+let todoChangePositionHandler = function(e){
+	enterHandler(e.key, handler.changeTodos, []);
+};
+
+//Enter event handler for change inputs
+let todoChangesInputs = document.querySelectorAll('.todoChangeInputs');
+let todoChangesInputsLength = todoChangesInputs.length;
+
+for(let i = 0; i < todoChangesInputsLength; i++){
+	todoChangesInputs[i].addEventListener("keypress", todoChangePositionHandler);
+}

```

Comment:
The event listener is added for both the input fields of "Change" functionality, which might call the corresponding handler on pressing enter on either one of input fields while the other being empty and thus causing problem in handler function running on null values.

Reply:
Yeah you are right, i will change that for the last change input only

Reply:
Changed this on commit f83c45a